csrc/executor_kernel_arg.cpp

```diff
@@ -14,6 +14,292 @@
 
 namespace nvfuser {
 
+namespace {
+
+// Forward traverse from rFactor domain to allocation domain, compute frontier
+// sizes and strides, validate that splits are divisible and merges are
+// contiguous, and update active_ids_ correspondingly.
+class ForwardTraverseFromRFactorToAlloc {
+  ExpressionEvaluator& ee_;
+  std::unordered_map<IterDomain*, std::pair<int64_t, int64_t>>& active_ids_;
+
+  void handle(Split* split) {
+    auto in = split->in();
+    auto inner = split->inner();
+    auto outer = split->outer();
+    auto in_it = active_ids_.find(in);
+    // TORCH_INTERNAL_ASSERT(in_it != active_ids_.end())
+    if (in_it == active_ids_.end()) {
+      // TODO: see [Allocation domain on both side of rFactor]
+      return;
+    }
+    auto [in_size, in_stride] = in_it->second;
+    auto factor = ee_.evaluate(split->factor())->as<int64_t>();
+    TORCH_INTERNAL_ASSERT(
+        in_size % factor == 0,
+        "non-divisible split is not allowed in allocation domain");
```

Comment:
Why isn't it allowed?

Reply:
This analysis will only run on tensors provided by PyTorch. PyTorch does not allow, for example, viewing a tensor of shape  `(1, 5)` as `(4,)`.

Reply:
I see. Can you add that to the error message? At first, I thought it's not allowed in nvfuser in general, which is not true.

Reply:
Added